libsolidity/formal/SMTChecker.cpp

```diff
@@ -597,19 +597,23 @@ void SMTChecker::checkCondition(
 	{
 	case smt::CheckResult::SATISFIABLE:
 	{
-		std::ostringstream message;
+		std::ostringstream message, modelMessage;
 		message << _description << " happens here";
 		if (m_currentFunction)
 		{
-			message << " for:\n";
+			modelMessage << "  for:\n";
 			solAssert(values.size() == expressionNames.size(), "");
+			map<string, string> sortedModel;
 			for (size_t i = 0; i < values.size(); ++i)
 				if (expressionsToEvaluate.at(i).name != values.at(i))
-					message << "  " << expressionNames.at(i) << " = " << values.at(i) << "\n";
+					sortedModel[expressionNames.at(i)] = values.at(i);
+
+			for (auto const& eval: sortedModel)
+				modelMessage << "  " << eval.first << " = " << eval.second << "\n";
 		}
 		else
 			message << ".";
-		m_errorReporter.warning(_location, message.str() + loopComment);
+		m_errorReporter.warning(_location, message.str() + loopComment, SecondarySourceLocation().append(modelMessage.str(), SourceLocation()));
```

Comment:
I just approved, but actually I think it would be cleaner to declare ``SecondarySourceLocation ssl;`` in the beginning of the scope and to only do the ``append`` inside the ``if``. ``modelMessage`` could be local to the ``if`` scope as well then.

Otherwise we end up with empty secondary source locations being added if ``!m_currentFunction``.